packages/web-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,194 @@
+   * Used to scale the tab active indicator up or down as animations of the active indicator occur.
+   */
+  private activeTabScale = 1;
+
+  /**
+   * appearance
+   * There are two modes of appearance: transparent and subtle.
+   */
+  @attr
+  public appearance?: TabsAppearance = TabsAppearance.transparent;
+
+  /**
+   * disabled
+   * Used for disabling all click and keyboard events for the tabs, child tab elements and tab panel elements. UI styling of content and tabs will appear as "grayed out."
+   */
+  @attr({ mode: 'boolean' })
+  public disabled?: boolean;
+
+  /**
+   * size
+   * Used to set the size of all the tab controls, which effects text size and margins. Three sizes: small, medium and large.
+   */
+  @attr
+  public size?: TabsSize = 'medium';
+
+  /**
+   * reserve-selected-tab-space
+   * Controls whether or not to make the text of the active tab bold. Bold text will cause a slight movement in the ui of the tabs. Defaults to false.
+   */
+  @attr({ attribute: 'reserve-selected-tab-space', mode: 'boolean' })
+  public reserveSelectedTabSpace?: boolean;
+
+  /**
+   * calculateAnimationProperties
+   *
+   * Recalculates the active tab offset and scale.
+   * These values will be applied to css variables that control the tab active indicator position animations
+   */
+  private calculateAnimationProperties(tab: Tab) {
+    this.activeTabOffset = this.getSelectedTabPosition(tab);
+    this.activeTabScale = this.getSelectedTabScale(tab);
+  }
+
+  /**
+   * getSelectedTabPosition - gets the x or y coordinates of the tab
+   */
+  private getSelectedTabPosition(tab: Tab): number {
+    if (this.orientation === 'horizontal') {
+      return this.previousActiveTabData.x - (tab.getBoundingClientRect().x - this.getBoundingClientRect().x);
+    } else return this.previousActiveTabData.y - (tab.getBoundingClientRect().y - this.getBoundingClientRect().y);
+  }
+
+  /**
+   * getSelectedTabScale - gets the scale of the
+   */
+  private getSelectedTabScale(tab: Tab): number {
+    if (this.orientation === 'horizontal') {
+      return this.previousActiveTabData.width / tab.getBoundingClientRect().width;
+    } else return this.previousActiveTabData.height / tab.getBoundingClientRect().height;
+  }
+
+  /**
+   * clearAnimationProperties
+   * resets animation values to defaults. removes the animated css class
+   */
+  private clearAnimationProperties(tab: Tab) {
+    this.previousActiveTabData = { id: '', x: 0, y: 0, height: 0, width: 0 };
+    this.activeTabOffset = 0;
+    this.activeTabScale = 1;
+    tab.classList.remove('animated');
+  }
+
+  /**
+   * animationLoop
+   * runs through all the operations required for setting the tab active indicator to its starting location, ending location, and applying the animated css class to the tab.
+   * @param tab
+   */
+  private animationLoop(tab: Tab) {
+    // If selected, set the animation properties.
+    if (tab.ariaSelected && tab.ariaSelected === 'true') {
```

Comment:
Where is the selected property? I don't see it on Tabs or Tab. Actually, I just realized I can remove this check anyway, since I'm only running the loop on the active tab. So the check is unecesarry.

We do need all this code in the loop. The animation loop only runs each time the active tab changes, and it does not run per tab.